internal/broker/broker_test.go

```diff
@@ -661,6 +669,50 @@ func TestEndSession(t *testing.T) {
 	require.NoError(t, err, "EndSession should not have returned an error when ending an existent session")
 }
 
+func TestUserPreCheck(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		username        string
+		allowedSuffixes []string
+
+		wantErr bool
+	}{
+		"Successfully allow user": {},
+		"Successfully allow user through multiple suffixes": {
```

Comment:
nitpick: I find "through multiple suffixes" a little bit confusing, why not "with multiple suffixes permitted" or something like this?